src/controllers/question.js

```diff
@@ -1,12 +1,102 @@
+const Question = require('../models/question');
+
+async function saveQuestion(req) {
+  let { question } = req;
+  question.title = req.body?.title;
+  question.content = req.body?.content;
+  question.subjects = req.body?.subjects;
+  question.isSolved = req.body?.isSolved;
+  try {
+    question.student = req.user?.id ?? null;
+    question = await question.save();
+    return question;
+  } catch (e) {
+    return { error: 'something went wrong' };
+  }
+}
+
 module.exports = {
   getAllQuestions: async () => {},
   getOneQuestion: async () => {},
-  addNewQuestion: async () => {},
-  updateQuestion: async () => {},
-  deleteQuestion: async () => {},
+  addNewQuestion: async (req, res) => {
+    try {
+      req.question = new Question();
+      const resp = await saveQuestion(req);
+      res.json(resp);
+    } catch (e) {
+      res.status(401).json('something went wrong');
+    }
+  },
+  updateQuestion: async (req, res) => {
+    const { id } = req.params;
+    try {
+      //  check authority for the user
+      if (id && req.user && req.user.id === id) {
```

Comment:
I think id here is the question id not user id, I'm not sure if we can write this condition (req.user.id === id) because its different ids,  maybe we need to fetch question details to find the related user then compare it like:
const question = await Question.findById(id);
if(question.student === req.user.id)

Reply:
done